File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/service/mapping/OptionalMapper.java
package com.github.arhor.simple.expense.tracker.service.mapping;

import java.util.Optional;

public final class OptionalMapper {

    public static <T> Optional<T> wrap(T value) {
        return Optional.ofNullable(value);
    }

    public static <T> T unwrap(Optional<T> value) {
        return value.orElse(null);
    }
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/mapping/OptionalMapper.kt
package com.github.arhor.simple.expense.tracker.service.mapping;

import java.util.*

object OptionalMapper {

    @JvmStatic
    fun <T : Any> wrap(value: T): Optional<T> = Optional.ofNullable(value)

    @JvmStatic
    fun <T> unwrap(value: Optional<T>): T? = value.orElse(null)
}
